From 5895caba6573e84f73f159d9e84cd1aa7e969f18 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 9 May 2021 14:37:00 -0700
Subject: [PATCH] replace pthread_mutexattr_setrobust_np with
 pthread_mutexattr_setrobust

This is now part of standard POSIX function [1]

Upstream-Status: Submitted [https://github.com/diamon/minicoredumper/pull/3]

[1] https://man7.org/linux/man-pages/man3/pthread_mutexattr_setrobust_np.3.html

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/minicoredumper_regd/daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/minicoredumper_regd/daemon.c b/src/minicoredumper_regd/daemon.c
index 115ec92..b9ad1ea 100644
--- a/src/minicoredumper_regd/daemon.c
+++ b/src/minicoredumper_regd/daemon.c
@@ -224,7 +224,7 @@ static int setup_shm(void)
 
 	pthread_mutexattr_init(&attr);
 	pthread_mutexattr_setprotocol(&attr, PTHREAD_PRIO_INHERIT);
-	pthread_mutexattr_setrobust_np(&attr, PTHREAD_MUTEX_ROBUST_NP);
+	pthread_mutexattr_setrobust(&attr, PTHREAD_MUTEX_ROBUST);
 	pthread_mutexattr_setpshared(&attr, PTHREAD_PROCESS_SHARED);
 	pthread_mutex_init(&sh->m, &attr);
 
-- 
2.31.1

